juju/juju
# Add abstraction for CLI filesystem access

## Description of change

When running as embedded CLI commands for the Dashboard, we need to disallow access to the filesystem.
This PR introduces a Filesystem interface on the base command struct, which delegates to the os pkg calls when the commands are run from a terminal, but when embedded the various functions return an error.

A new test is added to the juju/cmd package which parses all of the CLi code and fails if there are any calls to filesystem related functions from the os package. This includes assigning an os function to a variable for later use, and where an import alias may be used for the os pkg.

A followup PR will disable various flags which allow for file access so that the user isn't even given the option in the first place.

## QA steps

bootstrap
run commands which access the filesystem, eg
juju model-config somefile.yaml
juju deploy . --resource foo=bar.zip
juju upgrade-charm foo --path . --resource foo=baz.zip

Now run the commands embedded using https://gist.github.com/wallyworld/4e8de75d5a439bd511e27d9bf5046c31 and ensure they fail.
